File: layrz_sdk/entities/ats_entry.py
"""Entry entity"""

from pydantic import BaseModel, Field


class AtsEntry(BaseModel):
  """Entry entity"""

  pk: int = Field(description='Defines the primary key of the Function', alias='id')
  old_tank_level: float = Field(description='Old tank level in liters', default=0.0)
  new_tank_level: float = Field(description='New tank level in liters', default=0.0)
  density: float | None = Field(description='Density of the fuel in kg/m3', default=None)
  temperature: float | None = Field(description='Temperature of the fuel in Celsius', default=None)
  is_executed_by_command: bool = Field(description='Indicates if the entry is executed by command', default=False)

File: go/entities/ats_entry.go
package entities

// AtsEntry represents an entry entity in the ATS module.
type AtsEntry struct {
	// ID is the primary key of the entry.
	ID int64 `json:"id"`
	// OldTankLevel is the tank level in liters before the entry.
	OldTankLevel float64 `json:"old_tank_level"`
	// NewTankLevel is the tank level in liters after the entry.
	NewTankLevel float64 `json:"new_tank_level"`
	// Density is the density of the fuel in kg/m3.
	Density *float64 `json:"density,omitempty"`
	// Temperature is the temperature of the fuel in Celsius.
	Temperature *float64 `json:"temperature,omitempty"`
	// IsExecutedByCommand indicates if the entry is executed by command.
	IsExecutedByCommand bool `json:"is_executed_by_command"`
}
